File: frontend/src/types/network.ts
export type NetworkPredictResponse = {
  prediction: string
  confidence: number
  severity: string
  recommendation: string
  model_version: string
}

export type NetworkRowResult = {
  row: number
  prediction: string
  confidence: number
}

export type NetworkCsvResponse = {
  total_flows: number
  benign: number
  suspicious: number
  top_prediction: string
  results: NetworkRowResult[]
}

File: backend/src/anomaly_ai/schemas/network.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class NetworkPredictResponse(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    prediction: str
    confidence: float
    severity: str
    recommendation: str
    model_version: str


class NetworkRowResult(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    row: int
    prediction: str
    confidence: float


class NetworkCsvResponse(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    total_flows: int
    benign: int
    suspicious: int
    top_prediction: str
    results: list[NetworkRowResult]
